i6_models/assemblies/conformer/conformer.py

```diff
@@ -0,0 +1,179 @@
+        """
+        :param cfg: Conformer Block Configuration with subunits for the different Conformer parts
+        """
+        super().__init__()
+
+        self.ff_config = cfg.ff_cfg
+        self.mhsa_config = cfg.mhsa_cfg
+        self.conv_config = cfg.conv_cfg
+
+        self.ff_1 = ConformerPositionwiseFeedForwardV1(cfg=self.ff_config)
+        self.mhsa = ConformerMHSAV1(cfg=self.mhsa_config)
+        self.conv = ConformerConvolutionV1(cfg=self.conv_config)
+        self.ff_2 = ConformerPositionwiseFeedForwardV1(cfg=self.ff_config)
+        self.final_layer_norm = torch.nn.LayerNorm(self.ff_config.dim)
+
+    def forward(self, tensor: torch.Tensor):
+        """
+        :param tensor: input tensor of shape [B, T, F]
+        :return: torch.Tensor of shape [B, T, F]
+        """
+        residual = tensor  #  [B, T, F]
+        x = self.ff_1(tensor)  #  [B, T, F]
+        residual = 0.5 * x + residual  #  [B, T, F]
+        x = self.mhsa(residual)  #  [B, T, F]
+        residual = x + residual  # [B, T, F]
+        x = self.conv(residual)  #  [B, T, F]
+        residual = x + residual  # [B, T, F]
+        x = self.ff_2(residual)  #  [B, T, F]
+        x = 0.5 * x + tensor  #  [B, T, F]
+        x = self.final_layer_norm(x)  #  [B, T, F]
+        return x
+
+
+class ConformerFrontendV1(nn.Module):
+    """
+    Frontend part of the standard conformer doing down-sampling
+    """
+
+    def __init__(self, cfg: ConformerFrontendV1Config):
+        """
+        :param cfg: Conformer Frontend Configuration
+        """
+        super().__init__()
+
+        self.feature_dim = cfg.feature_dim
+        self.hidden_dim = cfg.hidden_dim
+        self.dropout_value = cfg.dropout
+        self.conv_stride = cfg.conv_stride
+        self.conv_kernel = cfg.conv_kernel
+        self.conv_padding = cfg.conv_padding
+
+        self.spec_aug = ...  # TODO
+        self.subsampling = nn.Conv1d(
+            in_channels=self.feature_dim,
+            out_channels=self.feature_dim,
+            kernel_size=self.conv_kernel,
+            stride=self.conv_stride,
+            padding=self.conv_padding,
+        )
+        self.linear = nn.Linear(in_features=self.feature_dim, out_features=self.hidden_dim)
+        self.dropout = nn.Dropout(p=self.dropout_value)
+
+    def forward(self, data_tensor: torch.Tensor):
+        """
+        :param data_tensor: input tensor of shape [B, T, F]
+        :return: torch.Tensor of shape [B, T', F']
+
+        F: feature dim after feature extraction, F': internal model feature dim
+        T: data time dim, T': down-sampled time dim
+        """
+        # TODO: SpecAugment
+        x = data_tensor.transpose(1, 2)  # [B, F, T]
+        x = self.subsampling(x)  # [B, F, T']
+        x = x.transpose(2, 1)  # [B, T', F]
+        x = self.linear(x)  # [B, T', F']
+        x = self.dropout(x)  # [B, T', F']
+
+        return x
+
+
+class ConformerEncoderV1(nn.Module):
+    def __init__(self, cfg: ConformerV1Config):
```

Comment:
```suggestion
    def __init__(self, cfg: ConformerEncoderV1Config):
```
Also here, I'd suggest to keep names for config and class the same.